### PR TITLE
refactor: extract cache package and logic of reading property value

### PR DESCRIPTION
**Pull request by genryxy**
Part of #87 

- Extracted `cache` package.
- Extracted logic of reading value of the property from environment and Artipie properties file.
- Fixed reading value of property and added tests for it. Now it takes value of property from environment, if it was set, otherwise it looks up in `artipie.properties` file. If both of them are `null` it parses to long passed default value.

### CODE REVIEW

File: src/main/java/com/artipie/misc/Property.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * Ctor.
+     * @param name Name of the property.
+     */
+    public Property(final String name) {
+        this.name = name;
+    }
+
+    /**
+     * Obtains long value of the property set properties or file
+     * with values of the properties.
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/misc/Property.java:31]: This sentence seems off, please, check

File: src/main/java/com/artipie/misc/Property.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+     * @param defval Default value for property
+     * @return Long value of or default value.
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/misc/Property.java:33]: and this one too

File: src/main/java/com/artipie/misc/Property.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+     * @throws ArtipieException In case of problem with parsing value of the property
+     */
+    public long asLongOrDefault(final String defval) {
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/misc/Property.java:36]: I'd suggest to change type of the parameter from String to long, as this method is designed to obtain and return longs

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy thanks, added some comments, take a look, please

**Comment by genryxy** [comment]
@olenagerasimova thanks, corrected

**Review by olenagerasimova — APPROVED** [review]
@genryxy thanks

**Comment by olenagerasimova** [comment]
@g4s8 please, take a look/merge

**Pull request merged**